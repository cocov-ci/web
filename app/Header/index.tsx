import Image from 'next/image';
import Link from 'next/link';
import styles from './Header.module.scss';

const Header = () => {
  return (
    <header className={styles.header}>
      <div className={styles.content}>
        <Image
          className={styles.logo}
          src='/images/logo.png'
          width='46'
          height='54'
          alt='Cocov logo'></Image>
        <p className={styles.title}>
          Cocov <span>v0.1 beta</span>
        </p>

        <nav>
          <ul className={styles.menu}>
            <li>
              <Link href='#'>Repositories</Link>
            </li>
            <li>
              <Link href='#'>Sign Out</Link>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  );
};

export default Header;
