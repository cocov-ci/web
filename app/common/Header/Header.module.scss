@import "styles/variables";

.header {
  border-bottom: 1px solid $colorGray;
  width: 100%;
  display: flex;
  justify-content: center;
  margin-bottom: 35px;

  .content {
    width: $contentWidth;
    height: 63px;
    display: flex;
    align-items: center;
    padding: 0 15px 0 6px;
  }

  .logo {
    display: inline-flex;
    align-items: flex-end;
    height: 100%;

    img {
      align-self: end;
    }
  }

  .title {
    flex: 1;
    font-size: $fontTitle;
    letter-spacing: -0.8px;
    margin-left: 10px;
    font-variation-settings: "wght" 700;

    span {
      font-size: initial;
      letter-spacing: initial;
      color: $colorGrayMedium;
      font-variation-settings: "wght" 400;
    }
  }

  .menu {
    display: inline-flex;
    list-style-type: none;

    li {
      font-size: 15px;
      padding-right: 16px;
      font-variation-settings: "wght" 500;
      cursor: pointer;

      &:hover {
        text-decoration: underline;
      }

      &:last-of-type {
        padding-right: 0;
      }
    }
  }
}
