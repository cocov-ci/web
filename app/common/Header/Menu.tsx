'use client'

import Link from 'next/link'
import { useEffect, useState } from 'react'

import useAuth from 'hooks/useAuth'
import { UserProps } from 'types/User'

import styles from './Header.module.scss'

const Menu = () => {
  const [loggedUser, setLoggedUser] = useState<UserProps>()
  const { logout, user } = useAuth()

  useEffect(() => {
    setLoggedUser(user)
  }, [user])

  if (!loggedUser) return null

  return (
    <nav>
      <ul className={styles.menu}>
        <li>
          <Link href="/">Repositories</Link>
        </li>
        {loggedUser?.isAdmin && (
          <li>
            <Link href="#">Adminland</Link>
          </li>
        )}
        <li>
          <p onClick={() => logout()}>Sign Out</p>
        </li>
      </ul>
    </nav>
  )
}

export default Menu
