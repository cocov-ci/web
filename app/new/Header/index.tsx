import SearchField from 'app/common/SearchField'
import Text from 'app/common/Text'

import styles from './Header.module.scss'
import LoadingComponent from './Loading'

interface HeaderProps {
  searchFieldDisabled?: boolean
  searchFieldLoading?: boolean
  loading: boolean
  onSearchChange: (arg: string) => void
}

const Header = ({
  searchFieldDisabled,
  searchFieldLoading,
  onSearchChange,
  loading,
}: HeaderProps) => {
  if (loading) {
    return <LoadingComponent />
  }

  return (
    <div className={styles.header}>
      <Text className={styles.description}>
        Below is a list of repositories from your GitHub Organization. Use the
        button at the side of the repository you wish to add to Cocov to add it
        to this instance. As the list may be quite large, the field on the right
        can be used to filter through repositories.
      </Text>
      <SearchField
        className={styles.searchField}
        disabled={searchFieldDisabled}
        loading={searchFieldLoading}
        onSearch={onSearchChange}
      />
    </div>
  )
}

export default Header
