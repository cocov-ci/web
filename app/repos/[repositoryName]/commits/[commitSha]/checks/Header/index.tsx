import Text from 'app/common/Text'
import CommitHeader from 'app/repos/[repositoryName]/commits/[commitSha]/CommitHeader'
import { HeadProps } from 'types/Commits'

import styles from './Header.module.scss'

export interface HeaderProps {
  repositoryName: string
  commit: HeadProps
}

const Header = ({ commit, repositoryName }: HeaderProps) => {
  return (
    <div className={styles.header}>
      <Text variant="title">
        <strong>Running Checks</strong>
      </Text>
      <Text variant="description">
        The list below shows which checks are being executed against the
        following commit:
      </Text>
      {commit && (
        <CommitHeader
          className={styles.commitHeader}
          head={commit}
          repositoryName={repositoryName}
        />
      )}
    </div>
  )
}

export default Header
