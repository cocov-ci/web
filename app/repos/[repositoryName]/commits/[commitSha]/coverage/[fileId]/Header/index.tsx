'use client'

import { X } from 'lucide-react'

import Kbd from 'app/common/Kbd'
import ProgressBar from 'app/common/ProgressBar'
import Text from 'app/common/Text'
import { FileIdReponseProps } from 'types/Coverage'

import styles from './Header.module.scss'

const Header = ({
  data,
  onClose,
}: {
  data: FileIdReponseProps
  onClose: () => void
}) => {
  if (!data) return null

  return (
    <div className={styles.header}>
      <div className={styles.data}>
        <div className={styles.file}>
          <Text variant="description">{data.file?.base_path}</Text>
          <Text variant="title">{data.file.name}</Text>
        </div>
        <div className={styles.stats}>
          <div className={styles.progressBar}>
            <Text variant="description">
              {data.coverage?.percent_covered}% covered
            </Text>
            <ProgressBar value={data.coverage.percent_covered} width="200px" />
          </div>
          <Text variant="description">
            {data.coverage.lines_total} relevant lines.{' '}
            <span className={styles.covered}>
              {data.coverage.lines_covered} lines covered
            </span>{' '}
            and{' '}
            <span className={styles.missed}>
              {data.coverage.lines_total - data.coverage.lines_covered} lines
              missed
            </span>
          </Text>
        </div>
      </div>
      <div className={styles.close}>
        <X onClick={() => onClose && onClose()} size={30} />
        <Kbd size="mini" text="esc" />
      </div>
    </div>
  )
}

export default Header
