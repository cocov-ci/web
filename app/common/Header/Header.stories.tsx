import type { Meta, StoryObj } from '@storybook/react'

import { AuthContext } from 'context/AuthContext'

import Header from './index'

const meta: Meta<typeof Header> = {
  title: 'Common/Header',
  component: Header,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof Header>

export const Default: Story = {
  args: {},
}

export const Authenticated: Story = {
  args: {},
}

Authenticated.decorators = [
  Story => (
    <AuthContext.Provider
      value={{
        login: () => null,
        logout: () => null,
        loading: false,
        isAuthenticated: true,
        user: {
          isAdmin: true,
          name: 'John Doe',
        },
      }}
    >
      <Story />
    </AuthContext.Provider>
  ),
]
