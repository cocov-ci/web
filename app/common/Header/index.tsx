import Image from 'next/image'
import Link from 'next/link'

import styles from './Header.module.scss'
import Menu from './Menu'

const Header = () => {
  return (
    <header className={styles.header}>
      <div className={styles.content}>
        <Link className={styles.logo} href="/">
          <Image
            alt="Cocov logo"
            height="54"
            src="/icons/logo.png"
            width="54"
          />
        </Link>
        <p className={styles.title}>
          <Link href="/">
            Cocov <span>v0.1 beta</span>
          </Link>
        </p>
        <Menu />
      </div>
    </header>
  )
}

export default Header
