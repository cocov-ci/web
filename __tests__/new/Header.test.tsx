import { render, screen } from '@testing-library/react'

import Header from 'app/new/Header'

import '@testing-library/jest-dom'

const args = {
  searchFieldDisabled: false,
  searchFieldLoading: false,
  onSearchChange: jest.fn(),
  loading: false,
}
describe('new/Header', () => {
  it('renders Header', () => {
    const { container } = render(<Header {...args} />)

    expect(
      screen.getByText(
        'Below is a list of repositories from your GitHub Organization. Use the button at the side of the repository you wish to add to Cocov to add it to this instance. As the list may be quite large, the field on the right can be used to filter through repositories.',
      ),
    ).toBeVisible()
    expect(container.querySelector('.searchField')).toBeVisible()
  })

  it('renders Header with "loading" param', () => {
    const { container } = render(<Header {...args} loading={true} />)

    expect(
      container.getElementsByClassName('react-loading-skeleton').length,
    ).toEqual(2)
  })

  it('renders Header snapshots', () => {
    const { container } = render(<Header {...args} />)

    expect(container).toMatchSnapshot()
  })
})
