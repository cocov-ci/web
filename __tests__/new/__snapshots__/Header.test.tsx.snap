// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new/Header renders Header snapshots 1`] = `
<div>
  <div
    class="header"
  >
    <p
      class="text description"
    >
      Below is a list of repositories from your GitHub Organization. Use the button at the side of the repository you wish to add to Cocov to add it to this instance. As the list may be quite large, the field on the right can be used to filter through repositories.
    </p>
    <div
      class="searchField searchField"
    >
      <div
        class="base light"
        style="width: 208px;"
      >
        <div
          class="inputWrapper input"
        >
          <div
            class="iconContainer"
          >
            <svg
              class="lucide lucide-search"
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="11"
                cy="11"
                r="8"
              />
              <line
                x1="21"
                x2="16.65"
                y1="21"
                y2="16.65"
              />
            </svg>
          </div>
          <input
            class="input withIcon"
            placeholder="Type to Search"
            type="text"
            value=""
          />
          <div
            class="loadingWrapper"
          >
            <div
              class="loading center tiny"
            >
              <img
                alt="A spinner indicating an ongoing operation"
                class="spinner"
                data-nimg="1"
                decoding="async"
                height="15"
                loading="lazy"
                src="/icons/spinner.svg"
                srcset="/icons/spinner.svg 1x, /icons/spinner.svg 2x"
                style="color: transparent;"
                width="15"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
